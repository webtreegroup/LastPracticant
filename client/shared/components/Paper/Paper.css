.paper {
    border-radius: var(--space-xs);
    padding: var(--space-xs);

    &__title {
        font-weight: normal;
        margin: var(--space-sm) 0 var(--space-lg);
        font-size: 24px;
        line-height: 24px;
        letter-spacing: 0.18px;
    }

    &_sizes_full {
        width: 100%;
        height: 100%;
    }

    &_sizes_small {
        width: 340px;
    }
}
