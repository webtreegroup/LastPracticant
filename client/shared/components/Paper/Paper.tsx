import './Paper.css';

import React, { FC, memo } from 'react';
import { Paper as MUIPaper } from '@material-ui/core';
import bem from 'bem-cn';

interface PaperProps extends React.HtmlHTMLAttributes<HTMLElement> {
    sizes?: 'small' | 'medium' | 'large' | 'full'
    theme?: 'dark' | 'light'
    title?: string
}

const block = bem('paper');

export const Paper: FC<PaperProps> = memo(
    ({
        children,
        className,
        title,
        sizes = 'full',
        theme = 'dark',
        ...props
    }) => (
        <MUIPaper
            className={block({ sizes, theme }).mix(className).toString()}
            {...props}
        >
            {title ? (
                <>
                    <h2 className={block('title')}>{title}</h2>
                    <section className={block('content')}>
                        {children}
                    </section>
                </>
            ) : children}
        </MUIPaper>
    ),
);
