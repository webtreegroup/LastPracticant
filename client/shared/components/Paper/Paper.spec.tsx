import React from 'react';
import { cleanup, render } from '@testing-library/react';
import { Paper } from './Paper';

afterEach(cleanup);

describe('Paper', () => {
    it('Render Paper component', () => {
        const { asFragment } = render(<Paper title="Hello Paper" sizes="full" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
