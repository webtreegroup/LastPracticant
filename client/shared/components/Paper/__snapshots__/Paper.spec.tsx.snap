// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper Render Paper component 1`] = `
<DocumentFragment>
  <div
    class="paper paper_sizes_full paper_theme_dark"
  >
    <h2
      class="paper__title"
    >
      Hello Paper
    </h2>
    <section
      class="paper__content"
    />
  </div>
</DocumentFragment>
`;
